src/app/profile/profile.component.ts

```diff
@@ -69,7 +87,20 @@ export class ProfileComponent {
   public updateProfile() {
     const profileData = { ...this.profileForm.value, resume: this.resumeUrl };
 
-    localStorage.setItem('profile', JSON.stringify(profileData));
+    this.userService
```

Comment:
pass the right type and avoid casting, and also always make sure to destroy any  subscription

Reply:
I am just curious, how do you know that the profile has been updated successfully?